### ISSUE TITLE
PGP: Mozilla Persona to go offline

### ISSUE THREAD

**Issue by mkash32** [privly/privly-organization#41]
In the [PGP wiki page](https://github.com/privly/privly-organization/wiki/PGP), it says Mozilla Persona will be used for identification. I took a look at their [webpage](https://developer.mozilla.org/en-US/Persona) and it is mentioned that persona services will be shut down from November 2016.
Are there any open source alternatives to Mozilla Persona available? ([LastPass](https://lastpass.com/) and [Okta](https://www.okta.com/) provide similar services but they aren't open source)

**Comment by smcgregor** [comment]
Hi @mkash32, good question. I updated the wiki to read:

> The PGP application can be considered to have two parts. First, we have the application that performs encryption and decryption for keys. Here we use [PGP (Pretty Good Privacy)](https://en.wikipedia.org/wiki/Pretty_Good_Privacy) with [OpenPGP.js](http://openpgpjs.org). Second, we have the system that determines which keys we trust. Here we previously used a hack on top of Mozilla Persona's authentication system, but since Mozilla is shutting down the project we need a different  solution.

There are many potential solutions that offer various security and UX properties. We previously selected Persona because it is a system that is most directly tied to email identities while also not requiring any manual management of keys. Users are terrible at managing keys. Unfortunately, Mozilla never got the Persona project to a finished project and never put any money behind marketing it. So they killed it.

Insofar as a replacement for Persona is concerned, you are welcome to propose any replacement you see fit. We can discuss the relative merits at that time. Generally going with a closed source project won't work because they generally require some trust in the identity service. We don't want to introduce another trusted parted to the system.

**Comment by mkash32** [comment]
Actually I wasn't able to understand how Persona fits into the PGP implementation. I read the documentation for persona but I wasn't able to make any connection. Could you give a hint as to how those two would be connected?

**Comment by mkash32** [comment]
Never mind, I found the information at https://github.com/privly/privly-applications/blob/experimental-pgp/pgp/protocol.md to be helpful in understanding the role of persona.

**Comment by smcgregor** [comment]
Yeah, I should note we figured out how to use Mozilla's identity infrastructure for signing PGP keys, but this is not the use-case Persona was built for. It turns out that identifying users for websites is very similar to identifying a user for PGP keys. It was a hack and we got burned by piggy backing on a solution we didn't control when they announced its EOL. We could potentially use Persona's software without Mozilla's support, but that would entail trusting the Privly Foundation to operate email verification and identity infrastructure -- which is something we want to avoid.
